workspaces/src/network/mod.rs

```diff
@@ -92,6 +101,40 @@ where
         let contract = self.create_tla_and_deploy(id.clone(), sk, wasm).await?;
         contract.into()
     }
+
+    async fn dev_deploy_project(&self, project_path: &str) -> anyhow::Result<Contract> {
+        let messages = cargo::build_cargo_project(fs::canonicalize(project_path)?)?;
+        // We find the last compiler artifact message which should contain information about the
+        // resulting .wasm file
+        let compile_artifact = messages
+            .iter()
+            .filter_map(|m| match m {
+                cargo_metadata::Message::CompilerArtifact(artifact) => Some(artifact),
+                _ => None,
+            })
+            .last()
+            .ok_or(anyhow!(
+                "Cargo failed to produce any compilation artifacts. \
+                 Please check that your project contains a NEAR smart contract."
+            ))?;
+        // The project could have generated many auxiliary files, we are only interested in .wasm files
+        let wasm_files = compile_artifact
+            .filenames
+            .iter()
+            .filter(|f| f.as_str().ends_with(".wasm"))
+            .collect::<Vec<_>>();
+        match wasm_files.as_slice() {
+            [] => Err(anyhow!(
+                "Compilation resulted in no '.wasm' target files. \
+                 Please check that your project contains a NEAR smart contract."
+            )),
+            [file] => self.dev_deploy(&fs::read(file.canonicalize()?)?).await,
```

Comment:
the file reading could be async here, and my intuition is that it's probably best to, although probably doesn't matter in most cases